geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientProxyMembershipID.java

```diff
@@ -176,149 +177,114 @@ public static ClientProxyMembershipID getClientId(DistributedMember member) {
     }
-    {
-      systemMemberId = sys.getDistributedMember();
-      try (HeapDataOutputStream hdos = new HeapDataOutputStream(256, KnownVersion.CURRENT)) {
-        if (systemMemberId != null) {
-          // update the durable id of the member identifier before serializing in case
-          // a pool name has been established
-          DurableClientAttributes attributes = systemMemberId.getDurableClientAttributes();
-          if (attributes != null && attributes.getId().length() > 0) {
-            ((InternalDistributedMember) systemMemberId).setDurableId(attributes.getId());
-          }
+    systemMemberId = sys.getDistributedMember();
+    try (HeapDataOutputStream hdos = new HeapDataOutputStream(256, KnownVersion.CURRENT)) {
+      if (systemMemberId != null) {
+        // update the durable id of the member identifier before serializing in case
+        // a pool name has been established
+        DurableClientAttributes attributes = systemMemberId.getDurableClientAttributes();
+        if (attributes != null && attributes.getId().length() > 0) {
+          ((InternalDistributedMember) systemMemberId).setDurableId(attributes.getId());
         }
-        DataSerializer.writeObject(systemMemberId, hdos);
-        client_side_identity = hdos.toByteArray();
-      } catch (IOException ioe) {
-        throw new InternalGemFireException(
-            "Unable to serialize identity",
-            ioe);
       }
-
-      system = sys;
+      DataSerializer.writeObject(systemMemberId, hdos);
+      client_side_identity = hdos.toByteArray();
+    } catch (IOException ioe) {
+      throw new InternalGemFireException("Unable to serialize identity", ioe);
     }
+    system = sys;
     return client_side_identity;
-
   }
 
   private ClientProxyMembershipID(int id, byte[] clientSideIdentity) {
-    boolean specialCase = Boolean.getBoolean(GeodeGlossary.GEMFIRE_PREFIX + "SPECIAL_DURABLE");
-    String durableID = this.system.getProperties().getProperty(DURABLE_CLIENT_ID);
+    this(clientSideIdentity, getUniqueId(id), systemMemberId);
+  }
+
+  private static int getUniqueId(final int id) {
+    final boolean specialCase =
+        Boolean.getBoolean(GeodeGlossary.GEMFIRE_PREFIX + "SPECIAL_DURABLE");
+    final String durableID = system.getProperties().getProperty(DURABLE_CLIENT_ID);
     if (specialCase && durableID != null && (!durableID.equals(""))) {
-      this.uniqueId = durable_synch_counter;
+      return durable_synch_counter;
     } else {
-      this.uniqueId = id;
+      return id;
     }
-    this.identity = clientSideIdentity;
-    this.memberId = systemMemberId;
   }
 
   public ClientProxyMembershipID() {}
 
-  public ClientProxyMembershipID(DistributedMember member) {
-    this.uniqueId = 1;
-    this.memberId = member;
+  public ClientProxyMembershipID(final DistributedMember member) {
+    this(null, 1, member);
     updateID(member);
```

Comment:
comment: I love what you did to chain the constructors. I see you had to introduce a private pure function `getUniqueId()` to help.

I find it hokey that in this method we set `identity` to `null`, only to turn around and call the (mutating) `updateId()` right after that.

I was about to suggest extracting the guts of `updateID()` into a pure function you could call here to do likewise, until I went and looked at `updateID()` and saw it mutates no fewer than four (4) member variables. Ugh.

Reply:
Yeah, I agree. I saw all that and said nope!